packages/jazz-tools/src/schema/subscribe.ts

```diff
@@ -0,0 +1,290 @@
+import { LocalNode, RawCoMap } from "cojson";
+import { cojsonInternals } from "cojson";
+import { Account } from "../exports.js";
+import { activeAccountContext } from "../implementation/activeAccountContext.js";
+import {
+  AnonymousJazzAgent,
+  ID,
+  RefsToResolve,
+  RefsToResolveStrict,
+  Resolved,
+  SchemaInit,
+  isRefEncoded,
+} from "../internal.js";
+import { CoMapInstance, CoMapSchema, CoMapSchemaClass } from "./coMap.js";
+
+type SubscribeListener<S extends CoMapSchema, R extends RefsToResolve<S>> = (
+  value: Resolved<S, R>,
+  unsubscribe: () => void,
+) => void;
+
+function createResolvablePromise<T>() {
+  let resolve!: (value: T) => void;
+  let reject!: (reason?: any) => void;
+
+  const promise = new Promise<T>((res, rej) => {
+    resolve = res;
+    reject = rej;
+  });
+
+  return { promise, resolve, reject };
+}
+
+type ResolvablePromise<T> = ReturnType<typeof createResolvablePromise<T>>;
+
+class Subscription {
```

Comment:
This is an abstraction to make it look like we can subscribe to values before the load.

It's an hack that should disappear after the LocalNode improvements